opm/simulators/timestepping/SimulatorReport.cpp

```diff
@@ -62,134 +65,151 @@ namespace Opm
+            os << " (Failed: " << failureReport->total_well_iterations << "; "
+               << 100.0*failureReport->total_well_iterations/n << "%)";
         }
+        os << std::endl;
+
+        n = total_linearizations + (failureReport ? failureReport->total_linearizations : 0);
+        os << "Overall Linearizations:       " << n;
+        if (failureReport) {
+            os << " (Failed: " << failureReport->total_linearizations << "; "
+               << 100.0*failureReport->total_linearizations/n << "%)";
+        }
+        os << std::endl;
+
+        n = total_newton_iterations + (failureReport ? failureReport->total_newton_iterations : 0);
+        os << "Overall Newton Iterations:    " << n;
+        if (failureReport) {
+            os << " (Failed: " << failureReport->total_newton_iterations << "; "
+               << 100.0*failureReport->total_newton_iterations/n << "%)";
+        }
+        os << std::endl;
+
+        n = total_linear_iterations + (failureReport ? failureReport->total_linear_iterations : 0);
+        os << "Overall Linear Iterations:    " << n;
+        if (failureReport) {
+            os << " (Failed: " << failureReport->total_linear_iterations << "; "
+               << 100.0*failureReport->total_linear_iterations/n << "%)";
+        }
+        os << std::endl;
     }
 
-    void SimulatorReport::reportParam(std::ostream& os)
+    void SimulatorReport::operator+=(const SimulatorReportSingle& sr)
     {
-        if ( verbose_ )
-        {
-            os << "/timing/total_time=" << total_time
-               << "\n/timing/pressure/total_time=" << pressure_time
-               << "\n/timing/transport/total_time=" << transport_time
-               << "\n/timing/newton/iterations=" << total_newton_iterations
-               << "\n/timing/linear/iterations=" << total_linear_iterations
-               << std::endl;
+        if (sr.converged) {
+            success += sr;
+        } else {
+            failure += sr;
         }
+        stepreports.push_back(sr);
     }
 
+    void SimulatorReport::operator+=(const SimulatorReport& sr)
+    {
+        success += sr.success;
+        failure += sr.failure;
+        stepreports.insert(stepreports.end(), sr.stepreports.begin(), sr.stepreports.end());
+    }
+
+    void SimulatorReport::reportFullyImplicit(std::ostream& os) const
+    {
+        success.reportFullyImplicit(os, &failure);
+    }
+
+    void SimulatorReport::fullReports(std::ostream& os) const
+    {
+        os << "  Time(day)  TStep(day)  Assembly    LSolve    LSetup    Update    Output WellIt Lins NewtIt LinIt Conv\n";
+        for (size_t i = 0; i < this->stepreports.size(); ++i) {
+            const SimulatorReportSingle& sr = this->stepreports[i];
+            os.precision(10);
+            os << std::defaultfloat;
+            os << std::setw(11) << unit::convert::to(sr.global_time, unit::day) << " ";
+            os << std::setw(11) << unit::convert::to(sr.timestep_length, unit::day) << " ";
```

Comment:
These might look a little strange for anyone running a `LAB` case (time reported in hours), but I assume those doing so are a low percentage of Flow's users.